src/storage/animationsSchema.ts

```diff
@@ -0,0 +1,449 @@
+		belowTokens: z.literal(true).optional(),
+		waitUntilFinished: z
+			.number()
+			.refine(...nonZero)
+			.optional(),
+		zIndex: z.number().optional(),
+		duration: z
+			.number()
+			.describe('The duration of the animationDataObject in milliseconds.')
+			.positive()
+			.optional(),
+		tint: hexColour
+			.describe('A hexadecimal colour code to give the animationDataObject a certain tint.')
+			.optional(),
+		rotate: z
+			.number()
+			.describe('An angle in degrees (°) to rotate the animationDataObject.')
+			.gt(-180)
+			.lte(180)
+			.refine(...nonZero)
+			.optional(),
+		opacity: z.number().describe('An opacity scaler from 0 to 1 (exclusive).').positive().lt(1).optional(),
+		mask: z.literal(true).optional(),
+		fadeIn: z
+			.number()
+			.refine(...nonZero)
+			.or(easingOptions.extend({ value: z.number() }).strict())
+			.optional(),
+		fadeOut: z
+			.number()
+			.refine(...nonZero)
+			.or(easingOptions.extend({ value: z.number() }).strict())
+			.optional(),
+		scale: z
+			.number()
+			.or(
+				z
+					.object({
+						min: z.number().or(z.object({ x: z.number(), y: z.number() })),
+						max: z.number().optional(),
+					})
+					.strict(),
+			)
+			.optional(),
+		wait: z
+			.number()
+			.or(
+				z
+					.object({
+						min: z.number().refine(...nonZero),
+						max: z.number().optional(),
+					})
+					.strict(),
+			)
+			.optional(),
+		delay: z
+			.number()
+			.or(
+				z
+					.object({
+						min: z.number().refine(...nonZero),
+						max: z.number().optional(),
+					})
+					.strict(),
+			)
+			.optional(),
+		scaleToObject: z
+			.number()
+			.or(
+				z
+					.object({
+						value: z.number().positive(),
+						uniform: z.literal(true).optional(),
+						considerTokenScale: z.literal(true).optional(),
+					})
+					.strict(),
+			)
+			.optional(),
+		spriteOffset: z
+			.object({
+				offset: vector2,
+				gridUnits: z.literal(true).optional(),
+				local: z.literal(true).optional(),
+			})
+			.strict()
+			.optional(),
+		size: z
+			.number()
+			.or(
+				z
+					.object({
+						value: z.number().positive(),
+						gridUnits: z.literal(true).optional(),
+					})
+					.strict(),
+			)
+			.optional(),
+		persist: z
+			.literal(true)
+			.or(
+				z
+					.object({
+						value: z.literal(true).optional(),
+						persistTokenPrototype: z.literal(true).optional(),
+					})
+					.strict()
+					.refine(...nonEmpty),
+			)
+			.optional(),
+		repeats: z
+			.number()
+			.min(1)
+			.int()
+			.or(
+				z
+					.object({
+						count: z.number().min(1),
+						delayMin: z.number(),
+						delayMax: z.number(),
+					})
+					.strict(),
+			)
+			.optional(),
+		moveTowards: easingOptions
+			.extend({
+				target: vector2.or(z.string()), // Also allows VisibleFoundryTypes but those aren't encodable in JSON
+			})
+			.strict()
+			.optional(),
+		filter: z
+			.object({
+				type: z.string(),
+				options: z.object({}), // Structure isn't documented?
```

Comment:
This is a Sequencer thing (2nd parameter): https://github.com/fantasycalendar/FoundryVTT-Sequencer/blob/5883cef8e2defe4b6ea0bedd29dfb13fc42ed44b/typings/types.d.ts#L382

Reply:
Its documented, but given it requires predicating on the `type` string I imagine Wasp just didnt want to deal with that in the TS file.

https://fantasycomputer.works/FoundryVTT-Sequencer/#/api/filter